presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaMetadata.java

```diff
@@ -269,22 +272,38 @@ private boolean isTablesEnumeratingTable(SchemaTableName schemaTableName)
         Session session = ((FullConnectorSession) connectorSession).getSession();
 
         Optional<Set<String>> tables = filterString(constraint, TABLE_NAME_COLUMN_HANDLE);
+        Stream<QualifiedTablePrefix> prefixStream;
         if (tables.isPresent()) {
-            return prefixes.stream()
+            prefixStream = prefixes.stream()
                     .flatMap(prefix -> tables.get().stream()
                             .map(table -> new QualifiedObjectName(catalogName, prefix.getSchemaName().get(), table)))
                     .filter(objectName -> metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
-                    .map(QualifiedObjectName::asQualifiedTablePrefix)
-                    .collect(toImmutableSet());
+                    .map(QualifiedObjectName::asQualifiedTablePrefix);
```

Comment:
does this case need to be changed? If tables are given in the `constraint` then it is expected that there are not that many of them.